crates/ext-processor/src/service.rs

```diff
@@ -250,21 +248,22 @@ impl BulwarkProcessor {
         );
         metrics::register_histogram!("combined_decision_score");
 
-        let redis_pool: Option<Arc<bb8::Pool<RedisConnectionManager>>> =
+        let redis_pool: Option<Arc<deadpool_redis::Pool>> =
             if let Some(redis_addr) = config.state.redis_uri.as_ref() {
-                let pool_size = config.state.redis_pool_size;
-                // TODO: better error handling instead of unwrap/panic
-                let connection_manager = RedisConnectionManager::new(redis_addr.as_str()).unwrap();
-                // TODO: make pool size configurable
-                let pool = bb8::Pool::builder()
-                    .max_size(pool_size)
-                    .build(connection_manager)
-                    .await
-                    .unwrap();
-                Some(Arc::new(pool))
+                let cfg = deadpool_redis::Config {
+                    url: Some(redis_addr.into()),
+                    connection: None,
+                    pool: Some(deadpool_redis::PoolConfig::new(
+                        config.state.redis_pool_size,
+                    )),
+                };
+                Some(Arc::new(
+                    cfg.create_pool(Some(deadpool_redis::Runtime::Tokio1))?,
+                ))
```

Comment:
It's okay for this failure to prefer the application from starting. We explicitly don't want to continue in a degraded condition because the software is likely to be running in a blue/green deployment configuration. Continuing despite the failure could result in a healthy service being replaced with an unhealthy service.